bin/sh/Makefile

```diff
@@ -57,16 +58,17 @@ mksyntax: mksyntax.c ${BUILD_TOOLS_META}
 .ORDER: nodes.c nodes.h
 nodes.h: .NOMETA
 nodes.c nodes.h: mknodes nodetypes nodes.c.pat
-	${BTOOLSPATH:U.}/mknodes ${.CURDIR}/nodetypes ${.CURDIR}/nodes.c.pat
+	${BTOOLSPATH:U.}/mknodes ${SRCDIR}/nodetypes ${SRCDIR}/nodes.c.pat
 
 .ORDER: syntax.c syntax.h
 syntax.h: .NOMETA
 syntax.c syntax.h: mksyntax
 	${BTOOLSPATH:U.}/mksyntax
 
 token.h: mktokens
-	sh ${.CURDIR}/mktokens
+	sh ${SRCDIR}/mktokens
 
+.if ${PROG} == "sh"
```

Comment:
Hm, this is a bit ugly, would be better with an explicit flag like how I cleaned things up for cheribsdtest.

Reply:
What do you suggest instead?
```
BUILDING_BIN_SH?=1
.if ${BUILDING_BIN_SH}
```
?

Reply:
rtld-elf32 etc just do `MK_TESTS:= no` which would work here, no change to sh's own Makefile needed (other than the necessary `.PARSEDIR` and `?=` bits above).

Reply:
Uh, those don't actually, that's what the `-debug` one does, though it seems redundant as RTLD uses `.if ${.CURDIR} == ${RTLD_ELF_DIR}`. So I would say `.if ${.CURDIR} == ${SRCDIR}` is the most appropriate thing to use here?